externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/schema/SchemaHelper.scala

```diff
@@ -41,6 +41,24 @@ object SchemaHelper {
    */
   final val YEAR_MONTH_INTERVAL = "YearMonthIntervalType"
 
+  /**
+   * Spark 3.2.0 DataType UserDefinedType's class name.
+   * And UDT API has been public since Spark 3.2.0.
+   * Return true if dt is subClass of UserDefinedType, which has been set to public
+   * since Spark 3.2.0
+   */
+  final private val isUserDefinedType = (dt: DataType) => {
+    val dataType = "DataType"
+    val userDefinedType = "UserDefinedType"
```

Comment:
Comparing with the simple class name of the class could be confused if `DataType` or `UserDefinedType ` IS exactly used as in user's defined type. 
It would be better to use the class name. `org.apache.spark.sql.types` `org.apache.spark.sql.types.UserDefinedType`

Reply:
Sure, it's best to do it.